### PR TITLE
Post about qrexec policy changes

### PR DESCRIPTION
**Pull request by marmarta**
@marmarek , here's the post
I will push the missing image in a separate pull request.

### CODE REVIEW

**Comment by marmarta** [comment]
@andrewdavidwong  , could you look this over? @marmarek already did his review

**Comment by marmarek** [comment]
ping @andrewdavidwong

**Comment by andrewdavidwong** [comment]
Sorry for the delay. Will have this done soon.

**Comment by andrewdavidwong** [comment]
Great article, @marmarta! I've pushed a revision for your and @marmarek's review.

I noticed that the article didn't have a title, so I chose a provisional one: "Qubes Architecture Next Steps: The New Qrexec Policy System". The "Qubes Architecture Next Steps:" part makes it clear that this is part of a series (preceded by the GUI Domain article), and "The New Qrexec Policy System" seems to be an apt description of the work this article describes. But please feel free to choose your own title if you'd prefer a different one!

We can update the filename (for the date and URL slug) once you're satisfied.

**Comment by andrewdavidwong** [comment]
(It won't let me select @marmarta as a reviewer, probably due to GitHub group memberships, but I was able to select you both as assignees.)

**Comment by marmarta** [comment]
I like the title @andrewdavidwong  proposes a lot, especially the idea of clearly specifying the series. And thanks for the revisions :)

File: 2020-05-20-qrexec.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
+---
+layout: post
+title: "Qubes Architecture Next Steps: The New Qrexec Policy System"
+categories: articles
+author: Marta Marczykowska-Górecka
```
Comment (marmarek) [inline 2020-05-20-qrexec.md:5]: This should be similar to the previous post in the series: Marek Marczykowski-Górecki, Marta Marczykowska-Górecka

File: 2020-05-20-qrexec.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+More information about socket services can be found [here](https://github.com/QubesOS/qubes-issues/issues/3912).
+
+## Notifications
+
+Creating a qrexec policy daemon opened another convenient opportunity: adding notifications for some policy actions. By default, notifications are shown only when a *deny* occurs, unless it was due to an *ask* action. (If the user was already asked for consent, it is not necessary to show the notification.) However, the new policy format allows for an optional `notify=no` parameter with *deny* rules. *Allow* rules support a `notify=yes` parameter that will show a notification on accepted qrexec calls.
+
+[![Screenshot of an allow and a deny notifications](/attachment/wiki/posts/qrexec1.png)](/attachment/wiki/posts/qrexec1.png)
+ 
+For example, this rule will cause notifications to be shown whenever a `VM.exec` service call is made from the `work` qube to the `personal` qube:
```
Comment (marmarek) [inline 2020-05-20-qrexec.md:213]: The actual existing call is `qubes.VMExec`

File: 2020-05-20-qrexec.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+```
+qubes.VM.exec * work personal allow notify=yes
```
Comment (marmarek) [inline 2020-05-20-qrexec.md:216]: And here too.

**Comment by andrewdavidwong** [comment]
> Thanks @andrewdavidwong for the revisions. Besides minor comments it looks great.
> Of course date needs adjustment when publishing.

Yes indeed.

> I'm not sure why two other posts shows up in this PR now.

Sorry about that. It happened when I added my own commit to this PR. Can I use a different push command to avoid that?

**Comment by marmarek** [comment]
> Sorry about that. It happened when I added my own commit to this PR. Can I use a different push command to avoid that?

Switching to the PR branch without merging to master should do the trick. Something like this:
```
git fetch https://github.com/marmarta/qubes-posts patch-1
git checkout -b marmarta-patch-1 FETCH_HEAD
...
```
I have a script for doing something like this, but I think it is also easily doable with `hub` tool, you can simply `hub checkout <URL TO THE PR>`

**Comment by andrewdavidwong** [comment]
Hm, I'm pretty sure that's what I did, though.

**Comment by andrewdavidwong** [comment]
Btw, the article has been published and announced across all our channels!

**Comment by marmarek** [comment]
> Hm, I'm pretty sure that's what I did, though.

The difference is `git fetch` vs `git pull`.

> Btw, the article has been published and announced across all our channels!

Thanks!

**Comment by andrewdavidwong** [comment]
I believe I did:

```
git checkout -b marmarta-patch-1 master
git pull https://github.com/marmarta/qubes-posts patch-1
```

Then, after my changes, I did:

```
git push ssh://git@github.com/marmarta/qubes-posts.git marmarta-patch-1:patch-1
```

And that's when the unrelated commits showed up in this PR.

So it was the `git pull` that "mixed" the commits from marmarta's branch with the unrelated commits from `master`?

**Comment by marmarek** [comment]
Yes, `git pull` is equivalent to `git fetch` + `git merge`, so it merged the fetched changes to the master branch you've started from.

**Pull request merged**